File: packages/server-php/src/CredentialStore.php
<?php

declare(strict_types=1);

namespace OpenPasskey\Server;

interface CredentialStore
{
    public function store(StoredCredential $cred): void;

    /**
     * @throws PasskeyError if not found
     */
    public function get(string $credentialId): StoredCredential;

    /**
     * @return StoredCredential[]
     */
    public function getByUser(string $userId): array;

    /**
     * @throws PasskeyError if not found
     */
    public function update(StoredCredential $cred): void;

    /**
     * @throws PasskeyError if not found
     */
    public function delete(string $credentialId): void;
}

File: packages/server-rails/lib/open_passkey/stores/credential_store.rb
# frozen_string_literal: true

module OpenPasskey
  module CredentialStore
    def store(credential)
      raise NotImplementedError
    end

    def get(credential_id)
      raise NotImplementedError
    end

    def get_by_user(user_id)
      raise NotImplementedError
    end

    def update(credential)
      raise NotImplementedError
    end

    def delete(credential_id)
      raise NotImplementedError
    end
  end
end
